### PR TITLE
Fix Terraform application status issue

### LINKED ISSUE #1599: Could not get components from an application by `status.components`
**Describe the bug**

Applying the application terraform-webapp oam-dev/samples#59, no components could be seen by `status.components`.

```
kubectl get application -o yaml
apiVersion: v1
items:
- apiVersion: core.oam.dev/v1beta1
  kind: Application
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"core.oam.dev/v1beta1","kind":"Application","metadata":{"annotations":{},"name":"terraform-webapp","namespace":"default"},"spec":{"components":[{"name":"website","properties":{"password":"PasefjsfdY123!","zone_id":"cn-beijing-i"},"type":"alibaba-website"}]}}
    creationTimestamp: "2021-05-03T13:22:55Z"
    generation: 1
    managedFields:
    - apiVersion: core.oam.dev/v1beta1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:annotations:
            .: {}
            f:kubectl.kubernetes.io/last-applied-configuration: {}
        f:spec:
          .: {}
          f:components: {}
      manager: k
      operation: Update
      time: "2021-05-03T13:22:55Z"
    - apiVersion: core.oam.dev/v1beta1
      fieldsType: FieldsV1
      fieldsV1:
        f:status:
          .: {}
          f:conditions: {}
          f:latestRevision:
            .: {}
            f:name: {}
            f:revision: {}
            f:revisionHash: {}
          f:rollout:
            .: {}
            f:batchRollingState: {}
            f:currentBatch: {}
            f:lastTargetAppRevision: {}
            f:rollingState: {}
            f:upgradedReadyReplicas: {}
            f:upgradedReplicas: {}
          f:status: {}
      manager: kubevela
      operation: Update
      time: "2021-05-03T13:22:56Z"
    name: terraform-webapp
    namespace: default
    resourceVersion: "68347841"
    selfLink: /apis/core.oam.dev/v1beta1/namespaces/default/applications/terraform-webapp
    uid: 44777c62-e1ff-43cb-adec-f3e3a50b603e
  spec:
    components:
    - name: website
      properties:
        password: PasefjsfdY123!
        zone_id: cn-beijing-i
      type: alibaba-website
  status:
    conditions:
    - lastTransitionTime: "2021-05-03T13:22:54Z"
      reason: Available
      status: "True"
      type: Parsed
    - lastTransitionTime: "2021-05-03T13:22:54Z"
      reason: Available
      status: "True"
      type: Built
    - lastTransitionTime: "2021-05-03T13:22:56Z"
      reason: Available
      status: "True"
      type: Applied
    - lastTransitionTime: "2021-05-03T13:22:56Z"
      message: 'app=terraform-webapp, comp=website, evaluate context error: invalid
        output of workload website: _|_ // value "output" not found'
      reason: ReconcileError
      status: "False"
      type: HealthCheck
    latestRevision:
      name: terraform-webapp-v1
      revision: 1
      revisionHash: cb8e53bbcdbccff2
    rollout:
      batchRollingState: ""
      currentBatch: 0
      lastTargetAppRevision: ""
      rollingState: ""
      upgradedReadyReplicas: 0
      upgradedReplicas: 0
    status: healthChecking
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

**To Reproduce**

Follow readme steps https://github.com/oam-dev/samples/pull/59/files#diff-581ea7feb97b57c66b5ad10b1c47976ff168261020de218a35fcb8d28f7eff68.

**Expected behavior**

**Screenshots**

**Cluster information**

**Additional context**

### PR DESCRIPTION
**Pull request by zzxwill**
Fix #1599

### CODE REVIEW

File: pkg/controller/core.oam.dev/v1alpha2/application/apply.go [hunk 1]
```diff
@@ -205,73 +206,102 @@ func (h *appHandler) statusAggregate(appFile *appfile.Appfile) ([]common.Applica
...
+	pCtx := process.NewContext(h.app.Namespace, wl.Name, appFile.Name, appFile.RevisionName)
+	if err := wl.EvalContext(pCtx); err != nil {
+		return nil, false, errors.WithMessagef(err, "app=%s, comp=%s, evaluate context error", appFile.Name, wl.Name)
+	}
+	return h.baseGetComponentStatus(pCtx, appFile, wl)
+}
+
+func (h *appHandler) getComponentStatusForTerraformModule(appFile *appfile.Appfile, wl *appfile.Workload) (*common.ApplicationComponentStatus, bool, error) {
+	pCtx := appfile.NewBasicContext(wl, appFile.Name, appFile.RevisionName, appFile.Namespace)
+	return h.baseGetComponentStatus(pCtx, appFile, wl)
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/apply.go:237]: why not create a new function for checking status of terraform module

**Review by wonderflow — COMMENTED** [review]
please try to fix the test coverage warnings

File: apis/core.oam.dev/v1beta1/register.go [hunk 2]
```diff
@@ -137,4 +149,5 @@ func init() {
 	SchemeBuilder.Register(&AppDeployment{}, &AppDeploymentList{})
 	SchemeBuilder.Register(&Cluster{}, &ClusterList{})
 	SchemeBuilder.Register(&ResourceTracker{}, &ResourceTrackerList{})
+	TerraformSchemeBuilder.Register(&terraformapi.Configuration{}, &terraformapi.ConfigurationList{})
```
Comment (zzxwill) [inline apis/core.oam.dev/v1beta1/register.go:152]: @wonderflow This seems odd. Any suggestions? Thanks.

File: go.mod [hunk 3]
```diff
@@ -86,6 +86,7 @@ replace (
 	// fix build issue https://github.com/docker/distribution/issues/2406
 	github.com/docker/distribution => github.com/docker/distribution v0.0.0-20191216044856-a8371794149d
 	github.com/docker/docker => github.com/moby/moby v17.12.0-ce-rc1.0.20200618181300-9dc6525e6118+incompatible
+	github.com/oam-dev/terraform-controller v0.1.4 => github.com/zzxwill/terraform-controller v0.1.5
```
Comment (zzxwill) [inline go.mod:89]: Will remove it after https://github.com/oam-dev/terraform-controller/pull/18 is merged.

File: apis/core.oam.dev/v1beta1/register.go (hunk 2 above)
Comment (zzxwill) [reply apis/core.oam.dev/v1beta1/register.go]: @wonderflow @wangkai1994 Done. Thanks.

**Review by wonderflow — APPROVED** [review]
good job, thanks

**Pull request merged**